### PR TITLE
Changes to support documentation versioning

### LINKED ISSUE #517: [website] Enable documentation versioning
Currently, the https://clusterlink.net website holds only the the latest (`main`) documentation. Thus, it may intermittently diverge from the last ("officially") release code. Despite the project being in the 0.x.y state, and allow breaking changes, we should see if we can enable version based documentation to match release artifacts.
There's already [Docys support](https://www.docsy.dev/docs/adding-content/versioning/) for this.

### PR DESCRIPTION
**Pull request by elevran**
Fixes #517 

See issue for discussion of options and step by step changes.
Whenever possible, each step from [issue comment](https://github.com/clusterlink-net/clusterlink/issues/517#issuecomment-2059584817) is done in a separate commit.

### CODE REVIEW

**Comment by elevran** [comment]
@michalmalka @kfirtoledo - this is ready for review.
@kfirtoledo, note that #507 and #530 would need to be rebased since the documentation moved from `docs` to `docs/main`

**Comment by elevran** [comment]
> 1. Can we remove the word Draft after the main?

certainly can. It is up to us and not part of the directory. Intentionally marked as `DRAFT` so users immediately know it is work in progress. Or we can make a note of that in the main/_index.md file (that's where the `DRAFT` is added anyway).

> 2. There is an option that the sidebar will show also the subdocuments like today (https://clusterlink.net/docs/)

Certainly. Hugo has options to 
- expand all entries all the time (can lead to long left sidebar - consider what would happen when we have >3 releases at the same time
- expand fixed number of sublevels for all (same issue as above, only less of a problem - we have two levels so the only remaining choice is to open 1 sublevel...)
- expand children, siblings and parents of current node (top level always shown). This is what https://etcd.io does and what is *currently implemented*.
- manually/programmatically generate the menus (too much work for too little value, IMHO)

> 3. What is the new process for every release, from the website's perspective?

See #531 and [this comment on #517](https://github.com/clusterlink-net/clusterlink/issues/517#issuecomment-2057109273)

**Comment by kfirtoledo** [comment]
From my side looks good, just decide what you want to change from the previous comments, also maybe move the release instruction to the developer documentation.
In addition, I tried the links, and  I got a page not found https://deploy-preview-528--clusterlink-net.netlify.app/docs/v0.1.0/concepts/fabric/peers.md (so need to fix it).

**Review by kfirtoledo — APPROVED** [review]

**Comment by elevran** [comment]
after some more experimentation:
- using file extensions (e.g., `users.md`) works in the markdown (locally and on GitHub) and in local hugo deployments. However, these links are not rendered correctly on Netlify (e.g., local hugo would translate it to `users/` but netlfiy links would still contain `.md` and hence they result in 404). Decision: markdown linking is less important, make website work. Seems that named with `relref` may be best.
- foldable works well. Clicking on the link text does the "right thing" and closes the other menu entries, whereas clicking on the arrow/fold sign **only** (not the text) leaves the other menu entries in the same (open/close) state 

I will push a fix for these

File: website/content/en/docs/main/concepts/fabric.md [hunk 1]
```diff
@@ -44,5 +44,10 @@ This command will create the CA files `cert.pem` and `key.pem` in a directory na
 ## Related tasks
 
 Once a `Fabric` has been created and initialized, you can proceed with configuring
- [peers]({{< ref "peers" >}}). For a complete, end to end, use case please refer to the
- [iperf tutorial]({{< ref iperf >}}).
+ [peers][concept-peer]. For a complete, end to end, use case please refer to the
+ [iperf tutorial][tutorial-iperf].
+
+[concept-peer]: peers.md
```
[the remark this answers is not in the document]
Comment (elevran) [reply website/content/en/docs/main/concepts/fabric.md]: changed in a later commit

File: website/content/en/_index.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ title: ClusterLink
 ---
 
 {{< blocks/cover image_anchor="smart" height="auto" color="secondary" >}}
- {{% clusterlink/intro logo="clusterlink_logo.png" buttontext="Get Started" buttonlink="docs/getting-started" %}}Securely connecting<br />applications across multi-cloud{{% /clusterlink/intro %}}
+ {{% clusterlink/intro logo="clusterlink_logo.png" buttontext="Get Started" buttonlink="docs/main/getting-started" %}}Securely connecting<br />applications across multi-cloud{{% /clusterlink/intro %}}
```
[the remark this answers is not in the document]
Comment (elevran) [reply website/content/en/_index.md]: correct. Will push for this entire file

File: website/content/en/docs/main/tutorials/iperf.md [hunk 3]
```diff
@@ -116,7 +116,7 @@ kubectl apply -f $IPERF3_FILES/iperf3-server/iperf3.yaml
     clusterlink create peer-cert --name server
     ```
 
-    For more details regarding fabric and peer see [core concepts](https://clusterlink.net/docs/concepts).
```
[the remark this answers is not in the document]
Comment (elevran) [reply website/content/en/docs/main/tutorials/iperf.md:119]: The use of direct (production) website URL was replaced by a relative link to `[concepts]: ../concepts/`

File: website/content/en/docs/main/tutorials/iperf.md [hunk 4]
```diff
@@ -387,7 +387,7 @@ spec:
 {{% /tab %}}
 {{< /tabpane >}}
 
-For more details regarding policy configuration, see [here](https://clusterlink.net/docs/concepts/policies).
```
[the remark this answers is not in the document]
Comment (elevran) [reply website/content/en/docs/main/tutorials/iperf.md:390]: ditto: replaced by `[concept-policy]: {{< relref "../concepts/policies" >}}` at the bottom

**Pull request merged**